src/deepsparse/clip/zeroshot_pipeline.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, List, Type
+
+import numpy as np
+from numpy import linalg as la
+from pydantic import BaseModel, Field
+
+from deepsparse.clip import CLIPTextInput, CLIPVisualInput
+from deepsparse.pipeline import BasePipeline, Pipeline
+from scipy.special import softmax
+
+
+__all__ = ["CLIPZeroShotInput", "CLIPZeroShotOutput", "CLIPZeroShotPipeline"]
+
+
+class CLIPZeroShotInput(BaseModel):
+    """
+    Input for the CLIP Zero Shot Model
+    """
+
+    image: CLIPVisualInput = Field(
+        description="Path to image to run zero-shot prediction on."
+    )
+    text: CLIPTextInput = Field(description="List of text to process")
+
+
+class CLIPZeroShotOutput(BaseModel):
+    """
+    Output for the CLIP Zero Shot Model
+    """
+
+    # TODO: Maybe change this to a dictionary where keys are text inputs
+    text_scores: List[Any] = Field(description="Probability of each text class")
```

Comment:
I think that a dictionary would be fine in this case. Can you give an example of what this data structure looks like?

Reply:
I was thinking having keys as image names and values as the predicted class. I think the lists are fine for now but we can address them in a follow-up, if needed.